### ISSUE TITLE
OpenMM 8 causing test failures

### ISSUE THREAD

**Issue by mattwthompson** [openforcefield/openff-evaluator#492]
```
>           integrator = simulation.context.getIntegrator()
E           AttributeError: '_Simulation' object has no attribute 'context'

../../../micromamba-root/envs/evaluator-test-env/lib/python3.8/site-packages/openmm/app/statedatareporter.py:251: AttributeError
```
https://github.com/openforcefield/openff-evaluator/actions/runs/4049461838/jobs/6965854892

I don't see anything in the [release notes](https://github.com/openmm/openmm/releases/tag/8.0.0).

**Comment by j-wags** [comment]
Root cause here is that Evaluator has a [`_Simulation` class](https://github.com/openforcefield/openff-evaluator/blob/9154134dc28dfe07604cc003b64af992e0fa3e76/openff/evaluator/protocols/openmm.py#L413-L416) which imperfectly duck-types the OpenMM `Simulation` class. In OpenMM 8 some code path changed to ask for the `.context` attribute and the `_Simulation` object failed the test

**Comment by mattwthompson** [comment]
Fixed by #495
